File: ExcAlgoritmoProgramacao_JS/Lista_Bee/1534_basico.js
import { question } from "readline-sync";

function main() {
    let m = question();

    while(m !== null && m !== ``) {
        let n = parseInt(m);
        let matriz = [];
        let mensagem = ``;

        if(n >= 3 && n <= 70) {
            for (let i = 0; i < n; i++) {
                let linha = [];
    
                for (let j = 0; j < n; j++) {
                    if(i === j && i + j === n - 1){
                        linha.push(2);
                    }else if(i === j) {
                        linha.push(1);
                    }else if(i + j === n - 1) {
                        linha.push(2);
                    }else {
                        linha.push(3);
                    }
        
                }
                matriz.push(linha);
        
            }
            for (let m = 0; m < matriz.length; m++) {
                mensagem += `${matriz[m].join(``)}` + `\n`;
        
            }
            console.log(mensagem.trim());
            console.log(``);
            
        }
        m = question();

    }
}

main();

File: ExcAlgoritmoProgramacao_JS/Lista_Bee/1534_basico.py
def main():
    m = get_Text()

    while m != '':
        if m is not None:
            try:
                n = int(m)
            except ValueError:
                m = get_Text()
                continue
            
            matriz = []
            mensagem = ''

            for i in range(n):
                linha = []

                for j in range(n):
                    if (i == j and i + j == n - 1) or (i + j == n - 1):
                        linha.append(2)
                    elif i == j:
                        linha.append(1)
                    else:
                        linha.append(3)
            
                matriz.append(linha)
        
            for linha in matriz:
                mensagem += ''.join(str(x) for x in linha) + '\n'
        
            print(mensagem.strip())
            print('')

        m = get_Text()

def get_Text():
    try:
        return input().strip()
    except EOFError:
        return ''

main()